Snowflake.Data/Client/SnowflakeDbConnection.cs

```diff
@@ -134,9 +189,16 @@ public Task CloseAsync(CancellationToken cancellationToken)
             }
             else
             {
-                if ((_connectionState != ConnectionState.Closed) && SfSession != null)
+                if (IsNonClosedWithSession())

```

Comment:
This looks similar to the above. Maybe some extract function?

Reply:
Again - baby steps. I don't want to do a huge refactor so everyone would be confused

Reply:
On the other hand, isn't it a code that you just introduce? :)

Reply:
Already refactored.